### PR TITLE
feat: instrument otel in sql based extractors

### PR DESCRIPTION
**Pull request by batrov**

### CODE REVIEW

File: plugins/extractors/clickhouse/clickhouse.go [hunk 1]
```diff
@@ -53,13 +55,27 @@ func New(logger log.Logger) *Extractor {
...
+	if err != nil {
 		return err
 	}
 
-	var err error
-	e.db, err = sql.Open("clickhouse", e.config.ConnectionURL)
+	driverName := "clickhouse"
+	if config.OtelEnabled {
+		driverName, err = otelsql.Register(driverName,
+			otelsql.AllowRoot(),
```
Comment (sudo-suhas) [inline plugins/extractors/clickhouse/clickhouse.go]: https://github.com/nhatthm/otelsql#allowroot-and-span-context

> To fully take advantage of otelsql, all database calls should be made using the *Context methods. Failing to do so will result in many orphaned traces if the AllowRoot() is used. By default, AllowRoot() is disabled and will result in otelsql not tracing the database calls if context or parent spans are missing.

Lets not have `AllowRoot()` enabled. We should modify the queries to take the context instead.

File: plugins/extractors/clickhouse/clickhouse.go [hunk 2]
```diff
@@ -53,13 +55,27 @@ func New(logger log.Logger) *Extractor {
...
+			otelsql.TraceRowsClose(),
+			otelsql.TraceRowsAffected(),
+			otelsql.WithSystem(semconv.DBSystemClickhouse),
+		)
+		if err != nil {
+			return fmt.Errorf("register clickhouse otelsql wrapper: %w", err)
+		}
+	}
+
+	e.db, err = sql.Open(driverName, e.config.ConnectionURL)
```
Comment (sudo-suhas) [inline plugins/extractors/clickhouse/clickhouse.go]: We should also record stats for the opened database:

https://github.com/nhatthm/otelsql#usage

> Optionally, you could record [database connection metrics](https://github.com/nhatthm/otelsql#database-connection-metrics) using the `otelsql.RecordStats()`. For example:

> ```go
>	// Connect to a Postgres database using the postgres driver wrapper.
>	db, err := sql.Open(driverName, dsn)
>	if err != nil {
>		return nil, err
>	}
>
>	if err := otelsql.RecordStats(db); err != nil {
>		return nil, err
>	}
>
>	return db, nil
> ```

**Review by sudo-suhas — CHANGES_REQUESTED** [review]

File: plugins/extractors/clickhouse/clickhouse.go (hunk 2 above)
Comment (haveiss) [inline plugins/extractors/clickhouse/clickhouse.go]: looks redundant, possible to extracting this out?

File: plugins/extractors/clickhouse/clickhouse.go (hunk 1 above)
Comment (batrov) [reply plugins/extractors/clickhouse/clickhouse.go]: updated

File: plugins/extractors/clickhouse/clickhouse.go (hunk 2 above)
Comment (batrov) [reply plugins/extractors/clickhouse/clickhouse.go]: updated

Comment (batrov) [reply plugins/extractors/clickhouse/clickhouse.go]: updated, moved to sqlutil

File: plugins/extractors/mysql/mysql_test.go [hunk 3]
```diff
@@ -52,7 +54,7 @@ func TestMain(m *testing.M) {
 	}
 	// exponential backoff-retry, because the application in the container might not be ready to accept connections yet
 	retryFn := func(resource *dockertest.Resource) (err error) {
-		db, err = sql.Open("mysql", "root@tcp("+host+")/")
+		db, err = sqlutil.OpenWithOtel("mysql", "root@tcp("+host+")/", semconv.DBSystemMySQL)
```
Comment (sudo-suhas) [inline plugins/extractors/mysql/mysql_test.go]: I don't think we need to open connection with Otel instrumentation in tests.

We can undo the similar change in test files.

File: plugins/sqlutil/sqlutils.go [hunk 4]
```diff
@@ -62,3 +65,23 @@ func BuildBoolMap(strList []string) map[string]bool {
...
+		return nil, fmt.Errorf("register %s otelsql wrapper: %w", driverName, err)
+	}
+
+	db, err = sql.Open(driverName, connectionURL)
+	if err != nil {
+		return nil, err
+	}
+
+	err = otelsql.RecordStats(db)
+	return db, err
```
Comment (sudo-suhas) [inline plugins/sqlutil/sqlutils.go]: We should always return either a valid result with nil err or zero value result with err. 
```suggestion
	if err := otelsql.RecordStats(db); err != nil {
		return nil, err
	}

	return db, nil
```

File: plugins/extractors/mysql/mysql_test.go (hunk 3 above)
Comment (batrov) [reply plugins/extractors/mysql/mysql_test.go]: I think that would make the test invalid, since the actual code is actually instrumenting otel

Comment (sudo-suhas) [reply plugins/extractors/mysql/mysql_test.go]: This code specifically, is used only to open a connection and ping the database as a way to check whether the docker container is up and running. And the tests would still be opening their own connection with otelsql wrapper when we call init.

Comment (batrov) [reply plugins/extractors/mysql/mysql_test.go]: Actually this is only for confirming the container health, so should be fine without otel. Let me update this

Comment (batrov) [reply plugins/extractors/mysql/mysql_test.go]: updated

**Review by sudo-suhas — APPROVED** [review]

**Pull request merged**